### PR TITLE
youtube-dl: 2018.03.10 -> 2018.03.14

### PR DESCRIPTION
**Pull request by magnetophon**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/youtube-dl/default.nix [hunk 1]
```diff
@@ -16,11 +16,11 @@ with stdenv.lib;
 buildPythonApplication rec {
 
   name = "youtube-dl-${version}";
```
Comment (dotlambda) [inline pkgs/tools/misc/youtube-dl/default.nix]: `pname`

Comment (magnetophon) [reply pkgs/tools/misc/youtube-dl/default.nix]: @dotlambda What do you mean?

Comment (dotlambda) [reply pkgs/tools/misc/youtube-dl/default.nix]: You can drop the `name` and set `pname = "youtube-dl". This makes it easier to override the version.

Comment (magnetophon) [reply pkgs/tools/misc/youtube-dl/default.nix]: I tried that, but it doesn't build:
```
error: undefined variable 'name' at /home/bart/source/nixpkgs/pkgs/tools/misc/youtube-dl/default.nix:22:53
```

Comment (dotlambda) [reply pkgs/tools/misc/youtube-dl/default.nix]: That's because `${name}` is used in the url. You'll have to replace it by `${pname}-${version}`.

Comment (magnetophon) [reply pkgs/tools/misc/youtube-dl/default.nix]: Thanks, done.

File: pkgs/tools/misc/youtube-dl/default.nix [hunk 2]
```diff
@@ -15,12 +15,12 @@
 with stdenv.lib;
 buildPythonApplication rec {
 
-  name = "youtube-dl-${version}";
-  version = "2018.03.10";
+  pname = "youtube-dl-${version}";
```
Comment (dotlambda) [inline pkgs/tools/misc/youtube-dl/default.nix]: No version in here

Comment (magnetophon) [reply pkgs/tools/misc/youtube-dl/default.nix]: Done

**Comment by dotlambda** [comment]
@GrahamcOfBorg build youtube-dl

**Pull request merged**